mitiq/zne/tests/test_zne.py

```diff
@@ -560,3 +573,74 @@ def qs_noisy_simulation(
 
     mitigated = execute_with_zne(circuit, qs_noisy_simulation)
     assert abs(mitigated) < 1000
+
+
+@pytest.mark.parametrize(
+    "noise_scaling_method",
+    [fold_gates_at_random, insert_id_layers, fold_global, fold_all],
+)
+@pytest.mark.parametrize(
+    "extrapolation_factory", [RichardsonFactory, LinearFactory]
+)
+@pytest.mark.parametrize(
+    "to_frontend",
+    [None, to_qiskit, to_braket, to_pennylane, to_pyquil, to_qibo],
+)
+def test_two_stage_zne(
+    noise_scaling_method, extrapolation_factory, to_frontend
+):
+    qreg = cirq.LineQubit.range(2)
+    cirq_circuit = cirq.Circuit(
+        cirq.H.on_each(qreg),
+        cirq.CNOT(*qreg),
+        cirq.CNOT(*qreg),
+        cirq.H.on_each(qreg),
+    )
+    if to_frontend is not None:
+        frontend_circuit = to_frontend(cirq_circuit)
+    else:
+        frontend_circuit = cirq_circuit
+
+    scale_factors = [1, 3, 5]
+    circs = scaled_circuits(
+        frontend_circuit, scale_factors, noise_scaling_method
+    )
+
+    assert len(circs) == len(scale_factors)
+
+    if to_frontend == to_braket:
+        assert all(isinstance(circ, braket.circuits.Circuit) for circ in circs)
+    elif to_frontend == to_qiskit:
+        assert all(isinstance(circ, qiskit.QuantumCircuit) for circ in circs)
+    elif to_frontend == to_pennylane:
+        assert all(
+            isinstance(circ, pennylane.tape.QuantumTape) for circ in circs
+        )
+    elif to_frontend == to_pyquil:
+        assert all(isinstance(circ, pyquil.quil.Program) for circ in circs)
+    elif to_frontend == to_qibo:
+        assert all(isinstance(circ, qibo.Circuit) for circ in circs)
+    else:
+        assert all(isinstance(circ, cirq.Circuit) for circ in circs)
```

Comment:
These tests are redundant since we already know our scaling method functions preserve input type. E.g. `fold_global` never converts a `qibo.Circuit` to anything but a `qibo.Circuit`. Since those [tests](https://github.com/unitaryfund/mitiq/blob/357f2c0342ee68b551843f319281586422a5d1be/mitiq/zne/scaling/tests/test_folding.py) cover this, let's leave it out from here for readability.

Reply:
I see, that makes sense. I'll remove these tests!